pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -103,6 +103,16 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 					break
 				}
 			}
+			if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionResourcesAvailable); cond != nil && cond.Status == corev1.ConditionFalse {
+				for _, volume := range rev.Spec.Volumes {
+					if secret := volume.VolumeSource.Secret; secret != nil {
+						_, err := c.secretLister.Secrets(rev.Namespace).Get(secret.SecretName)
+						if err != nil {
+							rev.Status.MarkSecretNotFound("The secret cannot be found")
```

Comment:
This seems like a constant string, why is it a parameter.
Also it is probably useful to include the name of the parameter.
Also you override the already set error if more than one secret is set incorrectly.
Finally, you presume that the error is `NotFound` which is not correct.

Reply:
Yeah we should log the error here. I'm sure if we should include the secret name or the error message itself in the condition message since they might be too long.